drift/lib/src/runtime/types/converters.dart

```diff
@@ -33,17 +33,47 @@ abstract class TypeConverter<D, S> {
 /// to SQL (and vice-versa).
 /// When a [BuildGeneralColumn.map] column (or a `MAPPED BY` constraint in
 /// `.drift` files) refers to a type converter that inherits from
-/// [JsonTypeConverter], it will also be used for the conversion from and to
+/// [JsonTypeConverterWithDifferentTypes], it will also be used for the conversion from and to
 /// JSON.
-mixin JsonTypeConverter<D, S> on TypeConverter<D, S> {
+mixin JsonTypeConverterWithDifferentTypes<D, S, J> on TypeConverter<D, S> {
```

Comment:
Hard to think of a better name here... Maybe `BaseJsonTypeConverter` could work?

Reply:
Maybe just `JsonTypeConverter2`?

Reply:
Sounds good 👍